enhancements/evidence-forwarding.md

```diff
@@ -0,0 +1,305 @@
+  - TBD
+creation-date: 2022-07-12
+last-updated: 2022-07-12
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - TBD
+---
+
+To get started with this template:
+1. **Pick a domain.** Find the appropriate domain to discuss your enhancement.
+1. **Make a copy of this template.** Copy this template into the directory for
+   the domain.
+1. **Fill out the "overview" sections.** This includes the Summary and
+   Motivation sections. These should be easy and explain why the community
+   should desire this enhancement.
+1. **Create a PR.** Assign it to folks with expertise in that domain to help
+   sponsor the process.
+1. **Merge at each milestone.** Merge when the design is able to transition to a
+   new status (provisional, implementable, implemented, etc.). View anything
+   marked as `provisional` as an idea worth exploring in the future, but not
+   accepted as ready to execute. Anything marked as `implementable` should
+   clearly communicate how an enhancement is coded up and delivered. If an
+   enhancement describes a new deployment topology or platform, include a
+   logical description for the deployment, and how it handles the unique aspects
+   of the platform. Aim for single topic PRs to keep discussions focused. If you
+   disagree with what is already in a document, open a new PR with suggested
+   changes.
+1. **Keep all required headers.** If a section does not apply to an
+   enhancement, explain why but do not remove the section. This part
+   of the process is enforced by the linter CI job.
+
+See ../README.md for background behind these instructions.
+
+Start by filling out the header with the metadata for this enhancement.
+
+# Evidence Forwarding
+
+## Summary
+
+Compliance audits usually require evidence that proves an environment is
+compliant. While the compliance operator provides check results as a Custom
+Resource, users an generate complete reports in an [Asset Reporting Format
+(ARF)](https://csrc.nist.gov/Projects/Security-Content-Automation-Protocol/Specifications/arf).
+These reports can be large and verbose. A single report for a small cluster can
+generate about 3 MB of data. This increase with each scan run, and with the
+amount of infrastructure being scanned.
+
+Historically, the Compliance Operator saves these reports to a physical volume.
+This works around saving the entire report in Custom Resource Definitions,
+which are bound to limitations of etcd.
+
+Unfortunately, physical volumes have their own short-comings. Reports stored in
+physical volumes still need to be fetched for evidence. Physical volumes can
+also be deleted like any other cluster resource.
+
+A improvement to the idea of putting evidence reports on disk would be to dump
+them into an object or file store. This has an added benefit of allowing
+administrators managing multiple clusters to put evidence in a single location.
+
```

Comment:
This summary misses the point of needing to provide different kinds of assets when dealing with an audit. e.g. it would be important to know what kubernetes objects were evaluated as part of a kube scan.

The ARF output has a lot of information about the filesystem (OVAL is an inventory gathering mechanism after all) so it does have relevant information about different aspects of the system even if it's in a really verbose way. e.g. the gathered entities such as files will have info about the modes for the files. So, besides being results, we could use that as evidence in a way.